apps/wallet-mobile/src/features/Receive/common/ShareQRCodeCard/ShareQRCodeCard.tsx

```diff
@@ -0,0 +1,176 @@
+import {useTheme} from '@yoroi/theme'
+import _ from 'lodash'
+import * as React from 'react'
+import {StyleSheet, TouchableOpacity, useWindowDimensions, View} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import QRCode from 'react-native-qrcode-svg'
+import Animated, {FadeInDown, FadeOutDown, Layout} from 'react-native-reanimated'
+import Share from 'react-native-share'
+import ViewShot, {captureRef} from 'react-native-view-shot'
+
+import {Spacer, Text} from '../../../../components'
+import {CaptureShareQRCodeCard} from '../CaptureShareQRCodeCard/CaptureShareQRCodeCard'
+import {mocks} from '../mocks'
+import {useStrings} from '../useStrings'
+
+type ShareProps = {
+  address?: string
+  title?: string
+  addressDetails?: AddressDetailsProps
+  isCopying?: boolean
+  onLongPress?: () => void
+}
+
+type AddressDetailsProps = {
+  address: string
+  stakingHash?: string
+  spendingHash?: string
+  title?: string
+}
+
+export const ShareQRCodeCard = ({address, title, isCopying, onLongPress}: ShareProps) => {
+  const [isSharing, setIsSharing] = React.useState<boolean>(false)
+  const strings = useStrings()
+  const ref: React.RefObject<ViewShot> = React.useRef(null)
+
+  const {styles, colors} = useStyles()
+
+  const shareImage = () => {
+    setIsSharing(true)
+  }
+
+  React.useEffect(() => {
+    if (isSharing) {
+      const captureAndShare = () => {
+        _.delay(async () => {
```

Comment:
this logic only exists to keep the "sharing screen" up until the SS is taken, if the product changes it to be what is on screen, it only requires the ref. and all the "delays" can be deleted.

Reply:
yes, we can remove the `_.delay` and some of it. I think we can refactor this logic more when adding the functionality as well

Reply:
yap the other delay inside can also be removed after the new design